### ISSUE TITLE
Strange thing with .SWC files

### ISSUE THREAD

**Issue by rcaze** [brian-team/brian2#644]
Look at the following code snippet

```python
import brian2 as br2
from brian2 import cm, um, ms, siemens, mV

# Set simple equations
gL = 0.0003*siemens/cm**2
EL = -65*mV
pas = '''
Im = gL * (EL - v) : amp/meter**2
I : amp (point current)
'''

# Create a neuron from a swc
test_MDL = 'pyr_simp.swc'
simpl_swc = br2.Morphology(test_MDL)
nrn_swc = br2.SpatialNeuron(morphology=simpl_swc, model=pas)
M_swc = br2.StateMonitor(nrn_swc, "v", record=True)

# Create a neuron with Brian2 constructor
simpl = br2.Soma(diameter=10*um)
simpl.dend = br2.Cylinder(diameter=1*um, x=100*um, y=0*um, z=0*um)
nrn = br2.SpatialNeuron(morphology=simpl, model=pas)
M = br2.StateMonitor(nrn, "v", record=True)

br2.run(10*ms)

print M_swc.v
print M.v
```

I would expect to obtain the same result in both cases (loading from the follwing swc file)

```
# This is a test swc file
1 1 0.0 0.0 0.0 10 -1
2 3 100 0.0 0.0 1 1
```

But I do not. I obtain nan for all values in the swc case. What is the problem?  I did some tests around and the nans do no happen if the swc is a single line defining a soma.

**Comment by mstimberg** [comment]
Yes, this is not the only problem with the SWC import (see #569, #494 and the notes in the wiki) -- please don't use it for anything serious right now... I am currently working on a rewrite of the `Morphology` class, based on Romain's notes (see wiki). 
As a quick fix for your issue, you can replace line 160 in `brian2/spatialneuron/morphology.py`:
```Python
spherical_soma = len([line.split()[1] == 1 for line in lines if line[0] != '#'])==1
```
with
```Python
spherical_soma = sum([line.split()[1] == '1' for line in lines if line[0] != '#'])==1
```

Note that you did not use the correct values, though: SWC files specify radii not diameters.

**Comment by rcaze** [comment]
I did the change and this does not seem to fix the problem. I still have nans.

**Comment by mstimberg** [comment]
Just tried it again (with latest master + your code and your SWC file) and it works. Are you sure that you are using the fixed file during your test?

**Comment by rcaze** [comment]
Ok it works. My bet, I used ```len``` instead of ```sum```. Thank you for this quick fix. Happy to help on the morpholgy.py. We are doing some cool work on it with a PhD in Imperial.

**Comment by mstimberg** [comment]
This should be fixed via #652 .
